src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs

```diff
@@ -89,5 +92,54 @@ public static void InitializationWithUriOnNonAsyncReaderTrows()
         {
             Assert.Throws<System.Net.WebException>(() => XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = false }));
         }
+
+        [Fact]
+        public static void SynchronizationContextCurrent_NotUsedForAsyncOperations()
+        {
+            Task.Run(() =>
+            {
+                var sc = new TrackingSynchronizationContext();
+                SynchronizationContext.SetSynchronizationContext(sc);
+
+                using (XmlReader reader = XmlReader.Create(new DribbleReadXmlAsyncStream(_dummyXml), new XmlReaderSettings { Async = true,  }))
+                {
+                    while (reader.ReadAsync().GetAwaiter().GetResult());
+                }
+
+                Assert.True(sc.CallStacks.Count == 0, "Sync Ctx used: " + string.Join(Environment.NewLine + Environment.NewLine, sc.CallStacks));
```

Comment:
Could you move this assert from the test to the sync context itself?

Or would that cause an exception on a TP thread?

Reply:
That would cause an exception to occur somewhere we don't want it to occur.